### ISSUE TITLE
opengraph template: Support relative paths in `images` front-matter.

### ISSUE THREAD

**Issue by Sandelinos** [gohugoio/hugo#10665]
On my site I have a folder structure like this:
```
content
└── category
    └── post-1
        ├── index.md
        └── post-1-picture.jpg
```

And I have been defining the `images` front-matter like below and building my templates assuming the paths are relative.
```
---
title: Post 1
images: [post-1-picture.jpg]
---
```

However when I tried adding opengraph metadata to my `<head>` using the internal template (https://gohugo.io/templates/internal/), I realized that the internal templates expect `images` to have paths relative to the root of the site instead of the page they're defined in. So the tag I end up with has `content="http://example.com/post-1-picture.jpg"` instead of `content="http://example.com/category/post-1/post-1-picture.jpg"`, which is the real path of the file.

The markdown rendering already works with relative image paths and I think it would be a good to support this in the internal metadata templates as well to keep things consistent.

If this feature isn't approved, then the documentation should be made clearer about the fact that paths relative to the site's root are necessary. Currently all the examples in https://gohugo.io/templates/internal/  don't have any directory separators in the `images:` entries and there is no tree showing the directory structure the examples work with.

**Comment by bapps2** [comment]
> If [page bundles](https://gohugo.io/content-management/page-bundles/) are used and the images array is empty or undefined, images with filenames matching *feature* or *cover*,*thumbnail* are used for image metadata
https://gohugo.io/templates/internal/#configure-open-graph

**Comment by oxivanisher** [comment]
I thought https://github.com/gohugoio/hugo/releases/tag/v0.121.0 will also fix this, but sadly it did not.

**Comment by jmooring** [comment]
This _was_ resolved in v0.121.0. Try it:

```
git clone --single-branch -b hugo-github-issue-10665 https://github.com/jmooring/hugo-testing hugo-github-issue-10665
cd hugo-github-issue-10665
rm -rf public && hugo && grep "og:image" public/posts/post-1/index.html
```
